### PR TITLE
Fixes when rewards is reset ads service should shut down

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/11035

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm Ads state is deleted when reseting Brave Rewards
- Confirm Confirmations state is deleted when reseting Brave Rewards
- Confirm Ads state is not deleted when toggling Ads on/off switch
- Confirm Ads is started and ads can be shown when disabling and then re-enabling ads 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
Linux and Windows failed CI, restarting.

CI failed for unrelated gn check and browser tests on other platforms

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -3906,6 +3907,11 @@ void RewardsServiceImpl::OnCompleteReset(
     const bool success) {
   resetting_rewards_ = false;
 
+  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile_);
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.cc:3910]: not sure if this block is needed as you already have the same call in `RewardsServiceImpl::CompleteReset`. Line 3889

Comment (tmancey) [reply components/brave_rewards/browser/rewards_service_impl.cc:3905]: Fixed

**Comment by tmancey** [comment]
CI failed for unrelated `gn_check` and rewards browser tests

**Comment by LaurenWags** [comment]
Discussed with @NejcZdovc and instead of testing on Nightly (which is blocked by https://github.com/brave/brave-core/pull/6248), this is being tested with PR build from https://bravesoftware.slack.com/archives/C5DT3GPCH/p1596464740328400

```
Brave | 1.12.106 Chromium: 84.0.4147.105 (Official Build) nightly (64-bit)
-- | --
Revision | a6b12dfad6663f13a7e16e9a42a6a4975374096b-refs/branch-heads/4147@{#943}
OS | macOS Version 10.14.6 (Build 18G3020)
```

* Verified test plan from description.
* Confirmed if rewards are reset, then `Brave-Browser-Nightly/Default/ads_service` folder is deleted.
* Confirmed if rewards are reset, then `Brave-Browser-Nightly/Default/rewards_service` folder is deleted.
* Confirmed re-enabling rewards creates these two folders again.
* Confirmed toggling Ads off did not remove `Brave-Browser-Nightly/Default/ads_service` or `Brave-Browser-Nightly/Default/rewards_service` folders.
* Confirmed no ads messages seen in logs when Ads were toggled off.
* Confirmed if Ads are toggled off and then the browser is restarted, ads are still off after restart.
* Confirmed Ads can be toggled on and ads messages start appearing in the logs. 
* Confirmed when Ads were toggled back on the ads panel information was not lost.
* Confirmed when re-enabling Ads that ads were shown based on max per hour setting.

**Pull request merged**